### PR TITLE
*: reformat according to STYLE.md

### PR DESCRIPTION
**Pull request by jordanlewis**
This is a large automated change produced by running
`crlfmt -ignore "pb.*.go" -tab 2 -w .`

Additionally, `crlfmt` has been added to `GLOCKFILE` and will run in`make check`, failing if any reformats are created. Currently, `crlfmt` only checks for and fixes incorrectly wrapped function declarations.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8823)

### CODE REVIEW

**Comment by tamird** [comment]
Gonna hold off on reviewing this until the end of code yellow, I think this is going to produce many merge conflicts otherwise.

---

Reviewed 2 of 2 files at r1.
Review status: 2 of 170 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[build/check-style.sh, line 148 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8823#-KQ1EIPFgxatB31X0l08:-KQ1EIPFgxatB31X0l09:b-avaeha) ([raw file](https://github.com/cockroachdb/cockroach/blob/d5cb82fcc432ded0ffb42aec4d9b54c000b63aee/build/check-style.sh#L148)):*
> ```Shell
> 
> TestCrlfmt() {
>   crlfmt -ignore "pb.go" .
> ```

nit: prefer single quotes when not interpolating

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8823#-:-KQ1EXh8egRbayrIVyzC:btnh7n9)*

**Comment by RaduBerinde** [comment]
I definitely agree on holding off on this until the end of code-yellow, it would probably cause a lot of issues during merging.

One thing is that I'm not sure if we should enforce always putting everything on a single line if it fits.. Here's an example:
```
Before:
func createTestClientForUser(
  t *testing.T,
  stopper *stop.Stopper,
  addr, user string,
  dbCtx client.DBContext,		
) *client.DB {


After:
func createTestClientForUser(
  t *testing.T, stopper *stop.Stopper, addr, user string, dbCtx 
 ) *client.DB {
```

IMO the before version is much more readable so we may want to leave it at the latitude of the human (just like `gofmt` lets you split lines if that's what you want to do). Not sure if that would be difficult to implement.

**Comment by paperstreet** [comment]
I'm abstaining on now vs after code-yellow.

@RaduBerinde The Before is a little more readable, but (for my money) it's only a _little_ more readable and the overhead of having to pick between the two every time you write a function (and potentially having a disagreement with a reviewer) isn't worth it. One of the big benefits of a style guide is eliminating bikeshedding over these sorts of marginal calls.

And fwiw, I don't think `gofmt` is a strong example. I _hate_ that it reformats `if err != nil { return err }`, but overall its strong opinions do more good than harm

**Comment by jordanlewis** [comment]
I'll hold off on this until code yellow is over. I agree with @paperstreet that the small gain in readability in that case is offset by the workflow benefit (less review back-and-forth over formatting concerns) and reduction in cognitive overhead (don't have to worry about formatting when writing code).

**Comment by RaduBerinde** [comment]
Sure, I see the argument for reducing choices. I'm ok with going with the black turtleneck.

Nice work on the tool BTW!

**Comment by jordanlewis** [comment]
The tool was written 95% by @paperstreet if that wasn't clear!

File: build/check-style.sh [hunk 1]
```diff
@@ -144,6 +144,10 @@ TestStaticcheck() {
   staticcheck ./...
 }
 
+TestCrlfmt() {
+  crlfmt -ignore "pb.go" .
```
Comment (dt) [inline build/check-style.sh]: `-tab 2`?

**Comment by RaduBerinde** [comment]
@jordanlewis - when Tobi is done with the end-of-split stuff, can you regenerate this and open a PR against `master`?

Also, could we make `crlfmt` also call `goimports` automatically (or add a script that runs both somewhere)? Just so we can easily replace the "on save" command in our editors without each of us having to write that script.

**Comment by RaduBerinde** [comment]
I wonder if we should do a similar treatment to to function calls:
```
short(arg1, arg2, arg3)
almostfits(
  arg1, arg2, arg3
)
superlong(
  arg1,
  arg2,
  arg3,
)
```
Of course this might get tricky if an argument is also long (e.g. a `func` def).

**Comment by dt** [comment]
@RaduBerinde Hm, since function *calls* can be passed things other than just identifiers -- expressions, literals, etc -- I think there's much more room for more subjective decisions w.r.t. readability and clarity than the signatures, so I imagine i might end up fighting such a formatter at times?

**Comment by jordanlewis** [comment]
Yes, I'm planning to regenerate this soon. I agree that it would be nice to have `crlfmt` call `goimports`, but it might be easier as you suggest to write a wrapper for both of them. I'll open an issue.

I think the function formatting probably needs to be bikeshedded in a different forum because it's a change to our existing style guide and not just a tools improvement.

File: build/check-style.sh (hunk 1 above)
Comment (dt) [inline build/check-style.sh]: ping (just want to check before you go through re-generating it)

Comment (jordanlewis) [inline build/check-style.sh]: ack

**Comment by RaduBerinde** [comment]
LGTM, this is great!  Just doublecheck with @tschottdorf before merging.

**Comment by tschottdorf** [comment]
No objections.

**Comment by RaduBerinde** [comment]
The change includes some `.pb.gw.go` files; we should ignore those as well.

**Comment by RaduBerinde** [comment]
^ which begs the question - why didn't the TeamCity build detect that generated files don't match?

**Comment by jordanlewis** [comment]
@RaduBerinde it is suspicious that `go generate` didn't detect changes. I re-ran and it failed that time: https://teamcity.cockroachdb.com/viewLog.html?buildId=26357&buildTypeId=Cockroach_UnitTests_Check&tab=buildLog#_focus=230&state=230

Besides that mildly concerning issue, I've redone this and added `pb.*.go` to the ignore list.

**Comment by tamird** [comment]
Reviewed 30 of 168 files at r2, 157 of 157 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[build/check-style.sh, line 148 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/8823#-KQ1EIPFgxatB31X0l08:-KTBYeXmx4KKCDzl1eLE:b-ii973o) ([raw file](https://github.com/cockroachdb/cockroach/blob/d5cb82fcc432ded0ffb42aec4d9b54c000b63aee/build/check-style.sh#L148)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> nit: prefer single quotes when not interpolating
</details>
once more, prefer single quotes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8823)*

**Comment by jordanlewis** [comment]
Review status: 176 of 177 files reviewed at latest revision, 2 unresolved discussions.

---

*[build/check-style.sh, line 148 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/8823#-KQ1EIPFgxatB31X0l08:-KTB_fXsRzASz18yp4rv:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d5cb82fcc432ded0ffb42aec4d9b54c000b63aee/build/check-style.sh#L148)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> once more, prefer single quotes.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8823)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8823)*

**Comment by paperstreet** [comment]
🎉

**Comment by andreimatei** [comment]
Were unwrapping changes intentional in this PR? The tool is forcing this diff:
```
-func (s *Store) maybeMergeTimestampCaches(
-   subsumingRng *Replica,
-   subsumedRng *Replica,
-) error {
+func (s *Store) maybeMergeTimestampCaches(subsumingRng *Replica, subsumedRng *Replica) error {
```

So it's forcing me to fill the line up to 100 chars. Based on the discussion in #9687, seems like this was probably not intended? Can we fix it?

**Comment by tamird** [comment]
Please file an issue in https://github.com/cockroachdb/crlfmt.

**Comment by RaduBerinde** [comment]
@andreimatei this exact issue was discussed above. See [my comment](https://github.com/cockroachdb/cockroach/pull/8823#issuecomment-242508360) and the discussion that followed.

**Pull request merged**